src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/AdmissionControlMetricsCollector.java

```diff
@@ -0,0 +1,166 @@
+            "com.sonian.elasticsearch.http.jetty.throttling.AdmissionController";
+
+    private final static String ADMISSION_CONTROL_SERVICE =
+            "com.sonian.elasticsearch.http.jetty.throttling.JettyAdmissionControlService";
+
+    public AdmissionControlMetricsCollector() {
+        super(sTimeInterval, "AdmissionControlMetricsCollector");
+        this.value = new StringBuilder();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void collectMetrics(long startTime) {
+        long startTimeMillis = System.currentTimeMillis();
+        try {
+            Class admissionController = Class.forName(ADMISSION_CONTROLLER);
+            Class jettyAdmissionControlService = Class.forName(ADMISSION_CONTROL_SERVICE);
+
+            Method getAdmissionController = jettyAdmissionControlService.getDeclaredMethod(
+                    "getAdmissionController",
+                    String.class);
+
+            Object globalJVMMP = getAdmissionController.invoke(null, GLOBAL_JVMMP);
+            Object requestSize = getAdmissionController.invoke(null, REQUEST_SIZE);
+
+            if(Objects.isNull(globalJVMMP) && Objects.isNull(requestSize)) {
+                return;
+            }
+
+            value.setLength(0);
+
+            Method getUsedQuota = admissionController.getDeclaredMethod("getUsedQuota");
+            Method getTotalQuota = admissionController.getDeclaredMethod("getTotalQuota");
+            Method getRejectionCount = admissionController.getDeclaredMethod("getRejectionCount");
+
+            if(!Objects.isNull(globalJVMMP)) {
+                value.append(PerformanceAnalyzerMetrics.getJsonCurrentMilliSeconds())
+                        .append(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor)
+                        .append(new AdmissionControlMetrics(
+                                GLOBAL_JVMMP,
+                                (long)getUsedQuota.invoke(globalJVMMP),
+                                (long)getTotalQuota.invoke(globalJVMMP),
+                                (long)getRejectionCount.invoke(globalJVMMP)
+                        ).serialize());
+            }
+
+            if(!Objects.isNull(requestSize)) {
+                value.append(PerformanceAnalyzerMetrics.getJsonCurrentMilliSeconds())
+                        .append(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor)
+                        .append(new AdmissionControlMetrics(
+                                REQUEST_SIZE,
+                                (long)getUsedQuota.invoke(requestSize),
+                                (long)getTotalQuota.invoke(requestSize),
+                                (long)getRejectionCount.invoke(requestSize)
+                        ).serialize());
+            }
+
+            saveMetricValues(value.toString(), startTime);
+        } catch (ClassNotFoundException e) {
+            LOG.debug("AdmissionControl not detected. Skipping AdmissionControlMetricsCollector");
```

Comment:
Can we update this to a similar check as `isMasterThrottlingFeatureAvailable` in  [MasterThrottlingMetricsCollector](https://github.com/opendistro-for-elasticsearch/performance-analyzer/blob/3c28ebdcd5cb2763056fc92145410ca9ebfea4b5/src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/MasterThrottlingMetricsCollector.java#L95) ?

Reply:
Done, related CR in **performance-analyzer-rca** https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/pull/574